src/main/kotlin/com/exactpro/th2/common/schema/strategy/fieldExtraction/impl/AnyMessageFieldExtractionStrategy.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.common.schema.strategy.fieldExtraction.impl
+
+import com.exactpro.th2.common.grpc.AnyMessage
+import com.exactpro.th2.common.grpc.Message
+import com.exactpro.th2.common.schema.message.toJson
+import com.exactpro.th2.common.schema.strategy.fieldExtraction.AbstractTh2MsgFieldExtraction
+
+class AnyMessageFieldExtractionStrategy : AbstractTh2MsgFieldExtraction() {
+    override fun parseMessage(message: com.google.protobuf.Message): Message {
+        check(message is AnyMessage) { "Message is not an ${AnyMessage::class.qualifiedName}: ${message.toJson()}" }
+        check(message.hasMessage()) { "Message doesn't contain a parsed message inside: ${message.toJson()}" }
```

Comment:
We will filter only parsed messages? What happened if users want to filter only by metadata in raw messages? Maybe we should return created message only with metadata?

Reply:
At the moment there's no filtering for raw messages, even in existing router for raw message batches. Such filtering would be implemented in task TH2-1503